python/psautohint/otfFont.py

```diff
@@ -652,24 +654,37 @@ def convertBezToT2(bezString):
     return t2Program
 
 
+def _run_tx(args, font_format):
+    try:
+        subprocess.check_call(["tx"] + args)
+    except subprocess.CalledProcessError as e:
+        raise FontParseError(e)
+    except OSError:
+        raise NotImplementedError("%s font format requires 'tx'" % font_format)
+
+
 class CFFFontData:
-    def __init__(self, path, is_otf):
+    def __init__(self, path, font_format):
         self.inputPath = path
-        self.is_otf = is_otf
+        self.font_format = font_format
 
-        if is_otf:
+        if font_format == "OTF":
             # It is an OTF font, we can process it directly.
             font = TTFont(path)
             if "CFF " not in font:
                 raise FontParseError("OTF font has no CFF table <%s>." % path)
         else:
-            # It is s CFF font, package it in an OTF font.
-            with open(path, "rb") as fp:
-                data = fp.read()
+            # Else, package it in an OTF font.
+            if font_format == "CFF":
+                with open(path, "rb") as fp:
+                    data = fp.read()
+            else:
+                with tempfile.NamedTemporaryFile() as fp:
+                    _run_tx(["-cff", "+b", "-std", path, fp.name], font_format)
```

Comment:
Enable `USE_TX` tests on AppVeyor as well. I want to make sure this call does not fail on Windows.
It won't hurt to enable them on Linux as well, since we've experienced quite interesting results on the afdko repository when using temporary files.

Reply:
Enabled on AppVeyor. I tried installing AFDKO on Linux initially but it failed due to some missing headers, did not investigate.

Reply:
Yeah, I also noticed NamedTemporaryFile doesn't always work with Windows. Sometimes I have to create one with delete=False, then close it, and then finally remove it manually after doing my things with it. Otherwise Windows seems to lock the open temp file and doesn't let the other subprocess write to it. Or I just skip NamedTemporaryFile use the low level tempfile.mkstemp directly.

Reply:
> I tried installing AFDKO on Linux initially but it failed due to some missing headers

because there are no manylinux1 wheels on PyPI so one has to build from source and you need the gcc-multilib to be able to compile 32 bit on a 64 bit host.

Reply:
I thought there is one now after the recent changes (I’m using `--pre`), but I guess I’m mistaken. 32 bit build is needed for old `autohintexe`, right? We could wait then until this one is really dead a buried.

Reply:
ah you're right there is a manylinux1 wheel, at least for 2.7 and 3.6:
https://pypi.org/project/afdko/2.8.0rc0/#files

Reply:
Hmm, installing afdko seems to be silently skipped https://travis-ci.org/adobe-type-tools/psautohint/jobs/424703851#L1105, @anthrotype any idea?